### ISSUE TITLE
Vim commands

### ISSUE THREAD

**Issue by luisdavim** [eschutz/atom-console#3]
Hi,

I think this could be a great replacement for a vim ex mode on atom. Could you add some alias to make feel more like vim's ex mode?
I thinking about stuff like:
* :w 
* :q 
* :wq
* :split
* :vsplit
* :tabnew
* :tabnext
* :tabprev

**Comment by eschutz** [comment]
Hi!
Thanks for the suggestion. I'm not much of a vim user so I need some clarification – would these be executed non-interactively, like in vim's ex mode, or immediately as in vim's normal mode?
From my understanding (or lack thereof, not being a vim user), when in ex mode you execute multiple commands while not being able to see their output. Would you like a feature like this, or just aliases to vim commands?

**Comment by eschutz** [comment]
A few of these I can add in an update, like `:wq` and `:split {filename}`, but in the meantime most of these you can implement yourself in the aliases file.
In your atom installation location, in `packages/atom-console/lib/command/aliases.json`, add:
```json
  ":w":"core:save",
  ":q":"application:quit",
  ":split":"pane:split-down-and-copy-active-item",
  ":vsplit":"pane:split-right-and-copy-active-item",
  ":tabnew":"application:new-file",
  ":tabnext":"pane:show-next-item",
  ":tabprev":"pane:show-previous-item"
```

**Comment by luisdavim** [comment]
Hi, thanks I think this would be a good starting point.
For the cases I mentioned the aliases are enough.
Is there a way of adding this tot the init.coffee file instead of having to modify a file from a package?
I was thinking of something like this:
https://github.com/hurrymaplelad/atom-alias-command

that can be set up like this:
```
atom.packages.onDidActivatePackage (pack) ->
  if pack.name == 'alias-command'
    aliasCommand = atom.packages.getLoadedPackage('alias-command').requireMainModule()

    aliasCommand 'package-control:install',
      orig: 'settings-view:install-packages-and-themes'

# Vim like commands
      aliasCommand 'w',
        orig: 'core:save'
      aliasCommand 'wall',
        orig: 'window:save-all'
      aliasCommand 'wall-qall',
        orig: ['window:save-all', 'window:close']
      aliasCommand 'q',
        orig: 'core:close'
      aliasCommand 'wq',
        orig: ['core:save', 'core:close']
      aliasCommand 'wqall',
        orig: ['core:save', 'window:close']
      aliasCommand 'qall',
        orig: 'window:close'
      aliasCommand ':',
        orig: 'go-to-line:toggle'
      aliasCommand '%',
        orig: 'find-and-replace:show'
      aliasCommand 'split',
        orig: 'pane:split-down'
      aliasCommand 'vsplit',
        orig: 'pane:split-left'
      aliasCommand 'diffthis',
        orig: 'split-diff:toggle'
      aliasCommand 'git-blame',
        orig: 'blame:toggle'
```

**Comment by eschutz** [comment]
Thanks for the feedback, that sounds like a great suggestion. I've been meaning to find a less clumsy way of aliasing commands and I haven't seen this before, this seems like just the way to do it.
I'll be sure to look into it, I think it would improve the user-friendliness of the package a lot.

**Comment by eschutz** [comment]
Hey @luisdavim, sorry for the wait but I finally implemented this as of version 0.4.0. I would have liked to have it done earlier but I've been a bit busy recently.
It works similarly to the system you linked. I added the `add-alias` atom-console command, and you can specify multiple atom-only commands to be executed under the alias by separating them with commas. Documented [here](https://github.com/eschutz/atom-console/blob/master/commands.md#built-in-commands).
For convenience I also aliased all the vim commands you listed (apart from `diffthis` and `git-blame`, those reference non-default packages).
I hope you enjoy it! :)
